test_pull_requests

```diff
@@ -862,8 +1058,9 @@ popd
       repo_to_pull_request.each do |repo, pull_request|
         pull_request_comments = get_comments(pull_request['number'], repo)
         test_prefix_comment = get_comment_with_prefix(pull_request['number'], repo, settings['test_prefix'], pull_request_comments)
-        if !test_prefix_comment || !(test_prefix_comment['body'] =~ /^#{settings['test_prefix']} Waiting/)
-          being_queued_comment = "#{settings['test_prefix']} Waiting: Determining build queue position"
+        fields = extract_bot_comment_fields(test_prefix_comment['body'], settings) if test_prefix_comment
+        if !test_prefix_comment || !([:wait_in_queue, :wait_queue_pos, :wait_stable_build].include? fields[:state])
```

Comment:
> I think because it doesn't need to be in this state to move on from any waiting state.

Ehh, that's really weak. If we want the comment to track the state, just because we currently have it coded to ignore this state is not a strong argument. If we are waiting on a downstream build, and we figure out that we're good to go and need to test, we *should* update the comment.